### PR TITLE
Add unified OWNERS for user-guide patches reviews

### PR DESCRIPTION
**Pull request by fmuyassarov**
Add unified OWNERS for user-guide patches reviews so that
user guide patch reviewers are not blocked to give lgtm & approve
labels. This also means reviewers and approvers should give lgtm
and approve only in their area of expertise (GitHub project of Metal3.io)
or in other words where they actually have reviewer or approval rights.
/cc @zaneb @dtantsur @furkatgofurov7 @andfasano @kashifest

### CODE REVIEW

**Review by Rozzii — COMMENTED** [review]
LGTM

File: docs/user-guide/src/OWNERS [hunk 1]
```diff
@@ -3,18 +3,25 @@
 # See https://github.com/metal3-io/cluster-api-provider-metal3/blob/main/OWNERS
```
Comment (furkatgofurov7) [inline docs/user-guide/src/OWNERS:3]: This needs to be removed

**Review by furkatgofurov7 — CHANGES_REQUESTED** [review]
Looks good apart minor nit, please check:

File: docs/user-guide/src/OWNERS (hunk 1 above)
Comment (fmuyassarov) [reply docs/user-guide/src/OWNERS]: thanks, fixed

**Review by furkatgofurov7 — COMMENTED** [review]
/lgtm

**Pull request merged**